safehttp/url.go

```diff
@@ -66,8 +68,21 @@ func (u URL) Port() string {
 //
 // Note that the path is stored in decoded form: /%47%6f%2f
 // becomes /Go/. A consequence is that it is impossible to tell
-// which slashes in the path were slashes in the raw URL and which
+// which slashes in the path were slashes in the rawURL and which
 // were %2f.
 func (u URL) Path() string {
 	return u.url.Path
 }
+
+// Parse parses a rawURL string into a URL structure.
+//
+// The rawURl may be relative (a path, without a host) or absolute (starting
+// with a scheme). Trying to parse a hostname and path without a scheme is
+// invalid but may not necessarily return an error, due to parsing ambiguities.
+func Parse(rawurl string) (*URL, error) {
```

Comment:
This will be used as `safehttp.Parse`. Maybe change it to `safehttp.ParseURL`?

Reply:
Would definitely be more intuitive for the user :)